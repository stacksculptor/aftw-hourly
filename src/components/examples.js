/* eslint-disable import/prefer-default-export*/
import * as Hello from './Hello/Hello.example';

export { Hello };
