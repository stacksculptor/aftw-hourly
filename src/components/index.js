import AddImages from './AddImages/AddImages';
import AuthorInfo from './AuthorInfo/AuthorInfo';
import BookingInfo from './BookingInfo/BookingInfo';
import Button from './Button/Button';
import CurrencyInput from './CurrencyInput/CurrencyInput';
import DateInput from './DateInput/DateInput';
import Discussion from './Discussion/Discussion';
import FilterPanel from './FilterPanel/FilterPanel';
import HeroSection from './HeroSection/HeroSection';
import Input from './Input/Input';
import LabeledField from './LabeledField/LabeledField';
import ListingCard from './ListingCard/ListingCard';
import ListingCardSmall from './ListingCardSmall/ListingCardSmall';
import LocationAutocompleteInput from './LocationAutocompleteInput/LocationAutocompleteInput';
import Map from './Map/Map';
import MapPanel from './MapPanel/MapPanel';
import Menu from './Menu/Menu';
import MobileFrame from './MobileFrame/MobileFrame';
import ModalInMobile from './ModalInMobile/ModalInMobile';
import NamedLink from './NamedLink/NamedLink';
import NamedRedirect from './NamedRedirect/NamedRedirect';
import OrderDetailsPanel from './OrderDetailsPanel/OrderDetailsPanel';
import OrderDiscussionPanel from './OrderDiscussionPanel/OrderDiscussionPanel';
import PageLayout from './PageLayout/PageLayout';
import PaginationLinks from './PaginationLinks/PaginationLinks';
import Promised from './Promised/Promised';
import RoutesProvider from './RoutesProvider/RoutesProvider';
import SearchResultsPanel from './SearchResultsPanel/SearchResultsPanel';
import StripeBankAccountToken from './StripeBankAccountToken/StripeBankAccountToken';

export {
  AddImages,
  AuthorInfo,
  BookingInfo,
  Button,
  CurrencyInput,
  DateInput,
  Discussion,
  FilterPanel,
  HeroSection,
  Input,
  LabeledField,
  ListingCard,
  ListingCardSmall,
  LocationAutocompleteInput,
  Map,
  MapPanel,
  Menu,
  MobileFrame,
  ModalInMobile,
  NamedLink,
  NamedRedirect,
  OrderDetailsPanel,
  OrderDiscussionPanel,
  PageLayout,
  PaginationLinks,
  Promised,
  RoutesProvider,
  SearchResultsPanel,
  StripeBankAccountToken,
};
