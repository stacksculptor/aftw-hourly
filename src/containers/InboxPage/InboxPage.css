@import '../../marketplace.css';

.mobileTopbar {
  /* Size */
  width: 100%;
  height: var(--topbarHeight);

  /* Layout for child components */
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  /* fill */
  background-color: var(--matterColorLight);

  /* shadows */
  box-shadow: none;

  @media (--desktopViewport) {
    display: none;
  }
}

.heading {

  /* Layout */
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 29px 24px 0 24px;

  border-top-width: 1px;
  border-top-color: var(--matterColorNegative);
  border-top-style: solid;
  background-color: var(--matterColorLight);
  box-shadow: var(--boxShadow);
}

.title {

  /* Font */
  @apply --marketplaceH1FontStyles;
  color: var(--matterColor);

  /* Layout */
  margin-top: 0;
  margin-bottom: 0;
  @media (--desktopViewport) {
    margin-top: 0;
    margin-bottom: 0;
  }
}

.tabs {
  padding-top: 11px;
}

.error {
  color: red;
  margin: 24px;
}

.noResults {
  margin: 0 24px;
}

.itemList {
  margin: 0;
  padding: 18px 24px;
}

.listItem {
  /* Layout */
  margin-bottom: 11px;

  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: var(--matterColorNegative);

  &:last-child {
    border-bottom: none;
  }
}

.itemLink {
  /* Layout */
  display: flex;
  padding-bottom: 11px;

  /* Remove text underline from this link */
  text-decoration: none;
  &:hover {
    text-decoration: none;
  }
}

.itemAvatar {
  margin-right: 8px;
  width: 43px;
  height: 43px;
}

.itemAvatarImage {
  border-radius: 50%;
}

.rowNotificationDot {
  width: 6px;
  height: 6px;
  margin-top: 10px;
  margin-right: 8px;
}

.notificationDot {
  color: var(--matterColorLight);

  width: 100%;
  height: 100%;
  border-radius: 50%;
  background-color: var(--failColor);
}

.itemInfo {
  /* Layout */
  flex-grow: 1;
  flex: 1;
  flex-direction: column;
}

.itemUsername {
  /* Font */
  @apply --marketplaceH4FontStyles;
  color: var(--matterColor);
  line-height: 18px;

  margin-top: 3px;
  margin-bottom: 0px;
  @media (--desktopViewport) {
    margin-top: 0;
    margin-bottom: 0;
  }
}

.itemTimestamp {
  /* Font */
  @apply --marketplaceH5FontStyles;

  margin-top: 0px;
  margin-bottom: 0px;
  @media (--desktopViewport) {
    margin-top: 4px;
    margin-bottom: 0;
  }
}

.itemState {
  /* Font */
  @apply --marketplaceH5FontStyles;

  margin-top: 4px;
  margin-bottom: 0px;
  @media (--desktopViewport) {
    margin-top: 4px;
    margin-bottom: 0;
  }
}

.pagination {
  margin: auto 24px 0 24px;
}

/* Transaction status affects to certain font colors and weights */

.namePending {
  font-weight: var(--fontWeightBold);
  margin-top: 1px;
  line-height: 18px;
  margin-bottom: 3px;
}

.timePending {
  color: var(--matterColor);
}

.stateRequested,
.statePending {
  font-weight: var(--fontWeightSemiBold);
  color: var(--attentionColor);
}

.stateAccepted {
  font-weight: var(--fontWeightSemiBold);
  color: var(--successColor);
}
.stateDeclined,
.stateDelivered {
  color: var(--matterColorAnti);
}

.nameRequested,
.nameDeclined,
.nameAccepted,
.nameDelivered {
  font-weight: var(--fontWeightMedium);
}

.timeRequested,
.timeDeclined,
.timeAccepted,
.timeDelivered {
  color: var(--matterColorAnti);
}
