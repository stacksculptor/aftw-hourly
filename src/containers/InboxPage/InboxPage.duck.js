import { reverse, sortBy } from 'lodash';
import { addEntities } from '../../ducks/sdk.duck';

// ================ Action types ================ //

export const FETCH_ORDERS_OR_SALES_REQUEST = 'app/InboxPage/FETCH_ORDERS_OR_SALES_REQUEST';
export const FETCH_ORDERS_OR_SALES_SUCCESS = 'app/InboxPage/FETCH_ORDERS_OR_SALES_SUCCESS';
export const FETCH_ORDERS_OR_SALES_ERROR = 'app/InboxPage/FETCH_ORDERS_OR_SALES_ERROR';

// ================ Reducer ================ //

const entityRefs = entities =>
  entities.map(entity => ({
    id: entity.id,
    type: entity.type,
  }));

const initialState = {
  fetchInProgress: false,
  fetchOrdersOrSalesError: null,
  transactionRefs: [],
};

export default function checkoutPageReducer(state = initialState, action = {}) {
  const { type, payload } = action;
  switch (type) {
    case FETCH_ORDERS_OR_SALES_REQUEST:
      return { ...state, fetchInProgress: true, fetchOrdersOrSalesError: null };
    case FETCH_ORDERS_OR_SALES_SUCCESS:
      return { ...state, fetchInProgress: false, transactionRefs: entityRefs(payload) };
    case FETCH_ORDERS_OR_SALES_ERROR:
      console.error(payload); // eslint-disable-line
      return { ...state, fetchInProgress: false, fetchOrdersOrSalesError: payload };

    default:
      return state;
  }
}

// ================ Action creators ================ //

const fetchOrdersOrSalesRequest = () => ({ type: FETCH_ORDERS_OR_SALES_REQUEST });
const fetchOrdersOrSalesSuccess = transactions => ({
  type: FETCH_ORDERS_OR_SALES_SUCCESS,
  payload: transactions,
});
const fetchOrdersOrSalesError = e => ({
  type: FETCH_ORDERS_OR_SALES_ERROR,
  error: true,
  payload: e,
});

// ================ Thunks ================ //

const sortedTransactions = txs =>
  reverse(
    sortBy(txs, tx => {
      return tx.attributes ? tx.attributes.lastTransitionedAt : null;
    })
  );

export const loadData = params =>
  (dispatch, getState, sdk) => {
    const { tab } = params;

    const onlyFilterValues = {
      orders: 'order',
      sales: 'sale',
    };

    const onlyFilter = onlyFilterValues[tab];
    if (!onlyFilter) {
      return Promise.reject(new Error(`Invalid tab for InboxPage: ${tab}`));
    }

    dispatch(fetchOrdersOrSalesRequest());

    const queryParams = {
      only: onlyFilter,
      include: ['provider', 'customer'],
    };

    return sdk.transactions
      .query(queryParams)
      .then(response => {
        const transactions = sortedTransactions(response.data.data);
        dispatch(addEntities(response));
        dispatch(fetchOrdersOrSalesSuccess(transactions));
        return response;
      })
      .catch(e => {
        dispatch(fetchOrdersOrSalesError(e));
        throw e;
      });
  };
