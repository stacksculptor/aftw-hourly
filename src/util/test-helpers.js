import React from 'react';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router';
import { Provider } from 'react-redux';
import configureStore from '../store';

export const TestProvider = props => {
  const store = configureStore();
  return (
    <IntlProvider locale="en">
      <BrowserRouter>
        <Provider store={store}>
          {props.children}
        </Provider>
      </BrowserRouter>
    </IntlProvider>
  );
};
