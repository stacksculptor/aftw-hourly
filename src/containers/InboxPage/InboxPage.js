import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import { Page } from '../../components';

const toPath = (filter, id) => {
  switch (filter) {
    case 'orders':
      return `/order/${id}`;
    case 'sales':
      return `/sale/${id}`;
    default:
      return `/conversation/${id}`;
  }
};

const InboxPage = props => {
  const { filter } = props;
  return (
    <Page title={`${filter} page`}>
      <ul>
        <li><Link to={toPath(filter, 1234)}>Single thread</Link></li>
      </ul>
    </Page>
  );
};

const { string } = PropTypes;

// TODO: should the filter an enum?
InboxPage.propTypes = { filter: string.isRequired };

export default InboxPage;
