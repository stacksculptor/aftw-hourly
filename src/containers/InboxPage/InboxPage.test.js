import React from 'react';
import { RoutesProvider } from '../../components';
import { renderShallow, renderDeep } from '../../util/test-helpers';
import {
  fakeIntl,
  createCurrentUser,
  createUser,
  createTransaction,
  createBooking,
} from '../../util/test-data';
import { InboxPageComponent, InboxItem } from './InboxPage';
import routeConfiguration from '../../routeConfiguration';
import { LINE_ITEM_NIGHT, TRANSITION_REQUEST } from '../../util/types';

const noop = () => null;

describe('InboxPage', () => {
  it('matches snapshot', () => {
    const provider = createUser('provider-user-id');
    const customer = createUser('customer-user-id');
    const currentUserProvider = createCurrentUser('provider-user-id');
    const currentUserCustomer = createCurrentUser('customer-user-id');

    const booking1 = createBooking('booking1', {
      start: new Date(Date.UTC(2017, 1, 15)),
      end: new Date(Date.UTC(2017, 1, 16)),
    });
    const booking2 = createBooking('booking2', {
      start: new Date(Date.UTC(2017, 2, 15)),
      end: new Date(Date.UTC(2017, 2, 16)),
    });

    const ordersProps = {
      unitType: LINE_ITEM_NIGHT,
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      params: {
        tab: 'orders',
      },
      authInProgress: false,
      currentUser: currentUserProvider,
      currentUserHasListings: false,
      isAuthenticated: false,
      fetchInProgress: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      transactions: [
        createTransaction({
          id: 'order-1',
          lastTransition: TRANSITION_REQUEST,
          customer,
          provider,
          lastTransitionedAt: new Date(Date.UTC(2017, 0, 15)),
          booking: booking1,
        }),
        createTransaction({
          id: 'order-2',
          lastTransition: TRANSITION_REQUEST,
          customer,
          provider,
          lastTransitionedAt: new Date(Date.UTC(2016, 0, 15)),
          booking: booking2,
        }),
      ],
      intl: fakeIntl,
      scrollingDisabled: false,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };

    const ordersTree = renderShallow(<InboxPageComponent {...ordersProps} />);
    expect(ordersTree).toMatchSnapshot();

    // Deeply render one InboxItem
    const orderItem = renderDeep(
      <InboxItem
        unitType={LINE_ITEM_NIGHT}
        type="order"
        tx={ordersProps.transactions[0]}
        intl={fakeIntl}
      />
    );
    expect(orderItem).toMatchSnapshot();

    const salesProps = {
      unitType: LINE_ITEM_NIGHT,
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      params: {
        tab: 'sales',
      },
      authInProgress: false,
      currentUser: currentUserCustomer,
      currentUserHasListings: false,
      isAuthenticated: false,
      fetchInProgress: false,
      onLogout: noop,
      onManageDisableScrolling: noop,
      transactions: [
        createTransaction({
          id: 'sale-1',
          lastTransition: TRANSITION_REQUEST,
          customer,
          provider,
          lastTransitionedAt: new Date(Date.UTC(2017, 0, 15)),
          booking: booking1,
        }),
        createTransaction({
          id: 'sale-2',
          lastTransition: TRANSITION_REQUEST,
          customer,
          provider,
          lastTransitionedAt: new Date(Date.UTC(2016, 0, 15)),
          booking: booking2,
        }),
      ],
      intl: fakeIntl,
      scrollingDisabled: false,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };

    const salesTree = renderShallow(<InboxPageComponent {...salesProps} />);
    expect(salesTree).toMatchSnapshot();

    // Deeply render one InboxItem
    const saleItem = renderDeep(
      <InboxItem
        unitType={LINE_ITEM_NIGHT}
        type="sale"
        tx={salesProps.transactions[0]}
        intl={fakeIntl}
      />
    );
    expect(saleItem).toMatchSnapshot();
  });
});
