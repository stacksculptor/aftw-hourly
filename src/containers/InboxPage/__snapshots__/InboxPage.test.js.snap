exports[`InboxPage matches snapshot 1`] = `
<Page
  authInfoError={null}
  logoutError={null}
  scrollingDisabled={false}
  title="InboxPage.ordersTitle">
  <Connect(withRouter(TopbarContainerComponent))
    currentPage="InboxPage" />
  <div>
    <div>
      <h1>
        <FormattedMessage
          id="InboxPage.title"
          values={Object {}} />
      </h1>
      <TabNav
        className={null}
        rootClassName={null}
        tabClassName={null}
        tabRootClassName={null}
        tabs={
          Array [
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "orders",
                },
              },
              "selected": true,
              "text": <span>
                <FormattedMessage
                      id="InboxPage.ordersTabTitle"
                      values={Object {}} />
          </span>,
            },
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "sales",
                },
              },
              "selected": false,
              "text": <span>
                <FormattedMessage
                      id="InboxPage.salesTabTitle"
                      values={Object {}} />
          </span>,
            },
          ]
        } />
    </div>
    <div>
      <ul>
        <li>
          <Component
            intl={
              Object {
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/preauthorize",
                  "lastTransitionedAt": 2017-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/night",
                      "lineTotal": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "unitPrice": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "lineTotal": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                      "unitPrice": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "amount": 900,
                    "currency": "USD",
                  },
                },
                "booking": Object {
                  "attributes": Object {
                    "end": 2017-02-16T00:00:00.000Z,
                    "start": 2017-02-15T00:00:00.000Z,
                  },
                  "id": UUID {
                    "uuid": "booking1",
                  },
                  "type": "booking",
                },
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "customer-user-id abbreviated name",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "uuid": "order-1",
                },
                "listing": null,
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "provider-user-id abbreviated name",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "type": "transaction",
              }
            }
            type="order" />
        </li>
        <li>
          <Component
            intl={
              Object {
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/preauthorize",
                  "lastTransitionedAt": 2016-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/night",
                      "lineTotal": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "unitPrice": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "lineTotal": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                      "unitPrice": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "amount": 900,
                    "currency": "USD",
                  },
                },
                "booking": Object {
                  "attributes": Object {
                    "end": 2017-03-16T00:00:00.000Z,
                    "start": 2017-03-15T00:00:00.000Z,
                  },
                  "id": UUID {
                    "uuid": "booking2",
                  },
                  "type": "booking",
                },
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "customer-user-id abbreviated name",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "uuid": "order-2",
                },
                "listing": null,
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "provider-user-id abbreviated name",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "type": "transaction",
              }
            }
            type="order" />
        </li>
      </ul>
    </div>
  </div>
</Page>
`;

exports[`InboxPage matches snapshot 2`] = `
<a
  className=""
  href="/order/order-1/details"
  onClick={[Function]}
  style={Object {}}
  title={null}>
  <div
    className={undefined}>
    <div
      className=""
      title="provider-user-id display name">
      <span
        className={undefined}>
        provider-user-id abbreviated name
      </span>
    </div>
  </div>
  <div
    className={undefined} />
  <div
    className={undefined}>
    <div
      className="">
      provider-user-id display name
    </div>
    <div
      className="">
      2017-02-15
       - 
      2017-02-16
      <span
        className={undefined}>
        10
      </span>
    </div>
  </div>
  <div
    className={undefined}>
    <div
      className="">
      InboxPage.stateRequested
    </div>
    <div
      className=""
      title="2017-01-15 00:00">
      2017-01-15
    </div>
  </div>
</a>
`;

exports[`InboxPage matches snapshot 3`] = `
<Page
  authInfoError={null}
  logoutError={null}
  scrollingDisabled={false}
  title="InboxPage.salesTitle">
  <Connect(withRouter(TopbarContainerComponent))
    currentPage="InboxPage" />
  <div>
    <div>
      <h1>
        <FormattedMessage
          id="InboxPage.title"
          values={Object {}} />
      </h1>
      <TabNav
        className={null}
        rootClassName={null}
        tabClassName={null}
        tabRootClassName={null}
        tabs={
          Array [
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "orders",
                },
              },
              "selected": false,
              "text": <span>
                <FormattedMessage
                      id="InboxPage.ordersTabTitle"
                      values={Object {}} />
          </span>,
            },
            Object {
              "linkProps": Object {
                "name": "InboxPage",
                "params": Object {
                  "tab": "sales",
                },
              },
              "selected": true,
              "text": <span>
                <FormattedMessage
                      id="InboxPage.salesTabTitle"
                      values={Object {}} />
          </span>,
            },
          ]
        } />
    </div>
    <div>
      <ul>
        <li>
          <Component
            intl={
              Object {
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/preauthorize",
                  "lastTransitionedAt": 2017-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/night",
                      "lineTotal": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "unitPrice": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "lineTotal": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                      "unitPrice": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "amount": 900,
                    "currency": "USD",
                  },
                },
                "booking": Object {
                  "attributes": Object {
                    "end": 2017-02-16T00:00:00.000Z,
                    "start": 2017-02-15T00:00:00.000Z,
                  },
                  "id": UUID {
                    "uuid": "booking1",
                  },
                  "type": "booking",
                },
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "customer-user-id abbreviated name",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "uuid": "sale-1",
                },
                "listing": null,
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "provider-user-id abbreviated name",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "type": "transaction",
              }
            }
            type="sale" />
        </li>
        <li>
          <Component
            intl={
              Object {
                "formatDate": [Function],
                "formatHTMLMessage": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatPlural": [Function],
                "formatRelative": [Function],
                "formatTime": [Function],
                "now": [Function],
              }
            }
            tx={
              Object {
                "attributes": Object {
                  "createdAt": 2017-05-01T00:00:00.000Z,
                  "lastTransition": "transition/preauthorize",
                  "lastTransitionedAt": 2016-01-15T00:00:00.000Z,
                  "lineItems": Array [
                    Object {
                      "code": "line-item/night",
                      "lineTotal": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                      "quantity": "1",
                      "unitPrice": Money {
                        "amount": 1000,
                        "currency": "USD",
                      },
                    },
                    Object {
                      "code": "line-item/provider-commission",
                      "lineTotal": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                      "unitPrice": Money {
                        "amount": -100,
                        "currency": "USD",
                      },
                    },
                  ],
                  "payinTotal": Money {
                    "amount": 1000,
                    "currency": "USD",
                  },
                  "payoutTotal": Money {
                    "amount": 900,
                    "currency": "USD",
                  },
                },
                "booking": Object {
                  "attributes": Object {
                    "end": 2017-03-16T00:00:00.000Z,
                    "start": 2017-03-15T00:00:00.000Z,
                  },
                  "id": UUID {
                    "uuid": "booking2",
                  },
                  "type": "booking",
                },
                "customer": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "customer-user-id abbreviated name",
                      "displayName": "customer-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "customer-user-id",
                  },
                  "type": "user",
                },
                "id": UUID {
                  "uuid": "sale-2",
                },
                "listing": null,
                "provider": Object {
                  "attributes": Object {
                    "banned": false,
                    "profile": Object {
                      "abbreviatedName": "provider-user-id abbreviated name",
                      "displayName": "provider-user-id display name",
                    },
                  },
                  "id": UUID {
                    "uuid": "provider-user-id",
                  },
                  "type": "user",
                },
                "type": "transaction",
              }
            }
            type="sale" />
        </li>
      </ul>
    </div>
  </div>
</Page>
`;

exports[`InboxPage matches snapshot 4`] = `
<a
  className=""
  href="/sale/sale-1/details"
  onClick={[Function]}
  style={Object {}}
  title={null}>
  <div
    className={undefined}>
    <div
      className=""
      title="customer-user-id display name">
      <span
        className={undefined}>
        customer-user-id abbreviated name
      </span>
    </div>
  </div>
  <div
    className={undefined}>
    <div
      className={undefined} />
  </div>
  <div
    className={undefined}>
    <div
      className="">
      customer-user-id display name
    </div>
    <div
      className="">
      2017-02-15
       - 
      2017-02-16
      <span
        className={undefined}>
        9
      </span>
    </div>
  </div>
  <div
    className={undefined}>
    <div
      className="">
      InboxPage.statePending
    </div>
    <div
      className=""
      title="2017-01-15 00:00">
      2017-01-15
    </div>
  </div>
</a>
`;
